internal/ui/util/util_test.go

```diff
@@ -26,4 +41,25 @@ var _ = Describe("Util", func() {
 			Expect(output).To(ContainSubstring("test\x1b[0m"))
 		})
 	})
+	Describe("NewStyleFromGradient", func() {
+		c1, _ := colorful.Hex("#ffffff")
+		c2, _ := colorful.Hex("#000000")
+		inputGradientFn := NewStyleFromGradient("#ffffff", "#000000")
+		When("the percent given is 100%", func() {
+			It("should generate text with the gradient of two colors relative to the percentage given", func() {
+				inputStyleFn := inputGradientFn(100)
+				output := inputStyleFn("test")
+				expectedOutput := NewStyle(c1.BlendHsv(c2, 1.0).Hex(), "", false)("test")
```

Comment:
This should be static text to guard against regressions in the NewStyle function with ANSI escape codes

Reply:
@achannarasappa, How should this be done? Can you give an example?

Reply:
Given the current behavior is the expected behavior, I would suggest changing the expected output to an empty string, running the test, copying the current output string (ANSI escape codes included) and pasting that into the expectes output as a hard coded string. Hope that makes some sense

Reply:
I set the `expectedOutput` to be an empty string like you suggested and ran the test, however, it doesn't show any ANSI escape codes in the failure but instead shows this:
```
Expected
          <string>: test
      to equal
          <string>: 
```

Tried making a separate file to print out the ANSI escape codes with similar code from the tests.
```go
package main

import (
	"fmt"

	. "github.com/achannarasappa/ticker/internal/ui/util"
)

func main() {
	inputGradientFn := NewStyleFromGradient("#ffffff", "#000000")
	output := inputGradientFn(100)("test")
	fmt.Printf("%#v", output)
}
```
Which gave an output of:
`"\x1b[38;2;0;0;0mtest\x1b[0m"`

When this value is copied and pasted in the tests:
```go
expectedOutput := "\x1b[38;2;0;0;0mtest\x1b[0m"
Expect(output).To(Equal(expectedOutput))
```
But it still fails:
```
Expected
          <string>: test
      to equal
          <string>: \x1b[38;2;0;0;0mtest\x1b[0m
```

It works when `expectedOutput` is changed to `"test"` or if the current method that this PR uses to test this is used.

Reply:
I've attempted to implement a fix I had in mind in [this commit](https://github.com/achannarasappa/ticker/pull/84/commits/c6fbbcab1611d2a61c40136d8942a80d9077590c). It worked locally but failed on CI so I've rolled back for now but will take a second look later today.

The essence of my suggestion was to assert the output equals static text as opposed to the output of another function. In this case that's not simple since the actual output varies by the terminal in use and the support color profile.

Reply:
👍 Thank you for looking into it.